libsyclinterface/tests/test_sycl_peer_access.cpp

```diff
@@ -0,0 +1,150 @@
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file has unit test cases for peer access functions defined in
+/// dpctl_sycl_device_interface.h.
+///
+//===----------------------------------------------------------------------===//
+
+#include "dpctl_sycl_device_interface.h"
+#include "dpctl_sycl_device_selector_interface.h"
+#include "dpctl_sycl_platform_interface.h"
+#include "dpctl_utils_helper.h"
+
+#include <gtest/gtest.h>
+#include <sycl/sycl.hpp>
+
+struct TestDPCTLPeerAccess : public ::testing::TestWithParam<const char *>
+{
+    DPCTLSyclPlatformRef P = nullptr;
+    DPCTLDeviceVectorRef DV = nullptr;
+
+    TestDPCTLPeerAccess()
+    {
+        auto DS = DPCTLFilterSelector_Create(GetParam());
+        if (DS) {
+            EXPECT_NO_FATAL_FAILURE(P = DPCTLPlatform_CreateFromSelector(DS));
+        }
+        DPCTLDeviceSelector_Delete(DS);
+        if (P) {
+            DV = DPCTLPlatform_GetDevices(P, DPCTLSyclDeviceType::DPCTL_ALL);
+        }
+    }
+
+    void SetUp()
+    {
+        if (!P || !DV) {
+            auto message = "Skipping as no devices of type " +
+                           std::string(GetParam()) + ".";
+            GTEST_SKIP_(message.c_str());
+        }
+
+        if (DPCTLDeviceVector_Size(DV) < 2) {
+            GTEST_SKIP_("Peer access tests require more than one device.");
+        }
+    }
+
+    ~TestDPCTLPeerAccess()
+    {
+        DPCTLDeviceVector_Delete(DV);
+        DPCTLPlatform_Delete(P);
+    }
+};
+
+TEST_P(TestDPCTLPeerAccess, ChkAccessSupported)
+{
+    auto D0 = DPCTLDeviceVector_GetAt(DV, 0);
+    auto D1 = DPCTLDeviceVector_GetAt(DV, 1);
+    ASSERT_TRUE(D0 != nullptr);
+    ASSERT_TRUE(D1 != nullptr);
+    EXPECT_NO_FATAL_FAILURE(DPCTLDevice_CanAccessPeer(
+        D0, D1, DPCTLPeerAccessType::access_supported));
+}
+
+TEST_P(TestDPCTLPeerAccess, ChkAtomicsSupported)
+{
+    auto D0 = DPCTLDeviceVector_GetAt(DV, 0);
+    auto D1 = DPCTLDeviceVector_GetAt(DV, 1);
+    ASSERT_TRUE(D0 != nullptr);
+    ASSERT_TRUE(D1 != nullptr);
+    EXPECT_NO_FATAL_FAILURE(DPCTLDevice_CanAccessPeer(
+        D0, D1, DPCTLPeerAccessType::atomics_supported));
+}
+
+TEST_P(TestDPCTLPeerAccess, ChkPeerAccess)
+{
+    auto D0 = DPCTLDeviceVector_GetAt(DV, 0);
+    auto D1 = DPCTLDeviceVector_GetAt(DV, 1);
+    ASSERT_TRUE(D0 != nullptr);
+    ASSERT_TRUE(D1 != nullptr);
+    bool canEnable = false;
+    EXPECT_NO_FATAL_FAILURE(canEnable = DPCTLDevice_CanAccessPeer(
+                                D0, D1, DPCTLPeerAccessType::access_supported));
+    if (canEnable) {
+        EXPECT_NO_FATAL_FAILURE(DPCTLDevice_EnablePeerAccess(D0, D1));
+        EXPECT_NO_FATAL_FAILURE(DPCTLDevice_DisablePeerAccess(D0, D1));
+    }
+}
+
+TEST_P(TestDPCTLPeerAccess, ChkPeerAccessToSelf)
+{
+    auto D0 = DPCTLDeviceVector_GetAt(DV, 0);
+    ASSERT_TRUE(D1 != nullptr);
+    bool canEnable = false;
+    EXPECT_NO_FATAL_FAILURE(canEnable = DPCTLDevice_CanAccessPeer(
```

Comment:
Do we need to add `ASSERT_FALSE(canEnable)` after that?

Reply:
I think it would be better not to test for a specific value, because the compiler might change soon.